### ISSUE TITLE
ID Site Logout callback url bug

### ISSUE THREAD

**Issue by mdeggies** [stormpath/stormpath-rails#78]
Hey Marko!

https://github.com/stormpath/stormpath-rails/blob/master/app/controllers/stormpath/rails/id_site_logout/new_controller.rb#L13

I think the callback URI here for the ID Site logout should be the same callback URI from the stormpath.yml file on line 13* (which is set by default to `/id_site_result`). 

Right now it's just using the root, so if you login with ID Site and then go to `http://localhost:3000/logout_id_site`, you'll get a 400:

> {"status":400,"code":2002,"message":"jwtBody cb_uri is an invalid value.","developerMessage":"jwtBody cb_uri is an invalid value.","moreInfo":"https://docs.stormpath.com/rest/product-guide/latest/errors.html#error-2002","requestId":"e6a58f10-d9d5-11e6-b155-22000a668cb0"}

On inspecting that JWT, the payload is this:

```
{
  iat: 1484342058,
  jti: "f1172ae3-0a57-47d6-a623-d58d521ac02c",
  iss: "74ITRU49J2PFFZU9KJDPDN103",
  sub: "https://api.stormpath.com/v1/applications/$ID",
  cb_uri: "http://localhost:3000/",
  path: "",
  state: ""
}
```

It should be the root + the cb_uri, so `http://localhost:3000/id_site_result` (unless I'm missing something).

**Comment by cilim** [comment]
Did you do a GET request to `http://localhost:3000/logout_id_site` or POST?

You usually don't hardcode the logout path, but rather use a path helper:

```ruby
<%= button_to 'Logout', logout_path, method: :post %>
```

**Comment by mdeggies** [comment]
Hey Marko,

I was originally following the docs here to log out of ID Site: https://docs.stormpath.com/ruby/product-guide/latest/idsite.html#step-3-optional-logging-out-of-id-site. This works perfectly. 

Then I saw that there was a route created for this in Rails -- ``logout_id_site``. There is no way to set the callback specifically for the ID site logout endpoint in the yaml configs: 

```
...
    idSite:
      enabled: true
      loginUri: ""
      forgotUri: "/#/forgot"
      registerUri: "/#/register"


    # A callback so Stormpath can pass information to the web application. This is
    # currently being used for ID Site, but may be used in the future for SAML,
    # Stormpath handled social login, webhooks, and other messages from Stormpath.
    callback:
      enabled: true
      uri: "/id_site_result"
...
```

But in the other integrations, the callback URI is the same for logout -- it should be``/id_site_result``. 

On line 13 here the callback for the ID Site logout route is set to the root url: https://github.com/stormpath/stormpath-rails/blob/master/app/controllers/stormpath/rails/id_site_logout/new_controller.rb#L13

Shouldn't it be root_url + callbackUri?

**Comment by cilim** [comment]
You are right Michele!

I'll start fixing the issue asap.

Thanks for reporting the issue!

**Comment by mdeggies** [comment]
To update, run ``bundle update stormpath-rails``! The newly released version 2.5.1 should resolve the bug.
